src/stp/plays/offensive/AttackingPass.cpp

```diff
@@ -35,58 +35,46 @@ AttackingPass::AttackingPass() : Play() {
+    passInfo = stp::computations::PassComputations::calculatePass(gen::AttackingPass, world, field);
 
-    calculateInfoForScoredRoles(world);
+    if (passInfo.passLocation == Vector2()) return 0;  // In case no pass is found
 
     // Score of play is the goalshotscore, adjusted based on the LoS and openness scores. The worse the LoS/Openness, the more the score is reduced
-    auto goalShotScore = static_cast<int>(PositionScoring::scorePosition(passLocation.value(), gen::GoalShot, field, world).score);
-    auto lineOfSightScore = static_cast<int>(PositionScoring::scorePosition(passLocation.value(), gen::LineOfSight, field, world).score);
-    auto openScore = static_cast<int>(PositionScoring::scorePosition(passLocation.value(), gen::Open, field, world).score);
+    auto goalShotScore = static_cast<int>(PositionScoring::scorePosition(passInfo.passLocation, gen::GoalShot, field, world).score);
+    auto lineOfSightScore = static_cast<int>(PositionScoring::scorePosition(passInfo.passLocation, gen::LineOfSight, field, world).score);
+    auto openScore = static_cast<int>(PositionScoring::scorePosition(passInfo.passLocation, gen::Open, field, world).score);
     return std::clamp(static_cast<int>(goalShotScore * (lineOfSightScore / 255.0) * (openScore / 255.0)), 0, 255);
 }
 
 Dealer::FlagMap AttackingPass::decideRoleFlags() const noexcept {
     Dealer::FlagMap flagMap;
-    Dealer::DealerFlag passerFlag(DealerFlagTitle::CLOSE_TO_BALL, DealerFlagPriority::REQUIRED);
-    Dealer::DealerFlag receiverFlag(DealerFlagTitle::WITH_WORKING_DRIBBLER, DealerFlagPriority::REQUIRED);
 
-    flagMap.insert({"keeper", {DealerFlagPriority::KEEPER, {}}});
-    flagMap.insert({"passer", {DealerFlagPriority::REQUIRED, {passerFlag}}});
-    flagMap.insert({"receiver", {DealerFlagPriority::HIGH_PRIORITY, {receiverFlag}}});
+    flagMap.insert({"keeper", {DealerFlagPriority::KEEPER, {}, passInfo.keeperId}});
+    flagMap.insert({"passer", {DealerFlagPriority::REQUIRED, {}, passInfo.passerId}});
+    flagMap.insert({"receiver", {DealerFlagPriority::HIGH_PRIORITY, {}, passInfo.receiverId}});
     flagMap.insert({"midfielder", {DealerFlagPriority::MEDIUM_PRIORITY, {}}});
 
     return flagMap;
 }
 
 void AttackingPass::calculateInfoForRoles() noexcept {
-    calculateInfoForScoredRoles(world);
-
     /// Keeper
     stpInfos["keeper"].setPositionToMoveTo(field.getOurGoalCenter());
     stpInfos["keeper"].setEnemyRobot(world->getWorld()->getRobotClosestToBall(world::them));
-    // TODO: set good pass position
-    stpInfos["keeper"].setPositionToShootAt(world->getWorld()->getRobotClosestToPoint(field.getOurGoalCenter(), world::us).value()->getPos());
-    stpInfos["keeper"].setKickOrChip(KickOrChip::KICK);
 
     /// Midfielder
     stpInfos["midfielder"].setPositionToMoveTo(PositionComputations::getPosition(std::nullopt, field.getMiddleMidGrid(), gen::SafePosition, field, world));
-}
 
-void AttackingPass::calculateInfoForScoredRoles(world::World* world) noexcept {
```

Comment:
You forgot to implement this function `calculateInfoForScoredRoles`. Compiling this branch results in an undefined reference. (It is a virtual function inherited from Play that is required to implement here)

Reply:
Whoops, had already fixed it but forgot to push. Should be better now